File: Leetcode Daily Challenge/032-February-2026/015-add-binary.java
// https://leetcode.com/problems/add-binary/




class Solution {
    public String addBinary(String a, String b) {
        int i = a.length() - 1, j = b.length() - 1, carry = 0;
        StringBuilder result =  new StringBuilder();
        while (i >= 0 || j >= 0 || carry > 0) {
            int sum = 0;
            if (i >= 0) {
                sum += a.charAt(i) - '0';
                i --;
            }
            if (j >= 0) {
                sum += b.charAt(j) - '0';
                j --;
            }
            sum += carry;
            result.append(sum % 2);
            carry = sum / 2;
        }
        return result.reverse().toString();
    }
}

File: Leetcode Daily Challenge/032-February-2026/015-add-binary.py
# https://leetcode.com/problems/add-binary/




class Solution:
    def addBinary(self, a: str, b: str) -> str:
        i, j, carry = len(a) - 1, len(b) - 1, 0
        result = []

        while i >= 0 or j >= 0 or carry > 0:
            total = carry
            if i >= 0:
                total += int(a[i])
                i -= 1
            if j >= 0:
                total += int(b[j])
                j -= 1

            result.append(str(total % 2))
            carry = total // 2
        
        return ''.join(reversed(result))
